sqf/analyzer.py

```diff
@@ -505,7 +505,9 @@ def execute_single(self, statement):
             outcome.position = base_tokens[0].position
 
         if statement.ending:
-            outcome = Nothing()
+            if isinstance(outcome, InterpreterType):
+                outcome = Nothing()
```

Comment:
There's a unit test checking the error position for `call {if x;}`, and without setting`Nothing` here, it would  see `call` as error position instead of `if`. 

This is where I'd have to dig really deep to understand why, and tbh I'm a bit too lazy right now.